config.toml.sample

```diff
@@ -0,0 +1,12 @@
+[github]
+# A GitHub personal access token.
+access_token = "<token>"
+
+# A GitHub oauth application for this instance of homu:
+app_client_id = "<id>"
+app_client_secret = "<secret>"
```

Comment:
In production I'd prefer we use environment variables for secrets rather than writing them in a TOML configuration file. That's the easiest way in our infrastructure to pass secrets into the container.

Options could be:

1. Do the same thing homu does, custom support for variable interpolation in the configuration file:

   ```toml
   access_token = "${GITHUB_TOKEN}"
   ```

2. Use a serde untagged enum, like Cargo does for version numbers:

   ```toml
   access_token = { env = "GITHUB_TOKEN" }
   ```

3. Keep the repository configuration in the file (maybe? see the GitHub App comment) but use only environmment variables (with `dotenvy` for local use) for secrets.

I'm personally leaning towards 3), preferring it over to 2). I don't think we should go with option 1) as it's more fragile.

Reply:
I assume that we will use a single, shared bors instance for all rust-XXX repositories, therefore passing lists in environment variables would probably be quite unwieldy. I'm not sure how 2) differs from 3). I assume that we would need to name the used environment variables in the config file and then pass them to the program? We could maybe auto-derive the env. name from the repository configuration in the file to avoid even mentioning the env. variable in the file.

Reply:
Yeah we'll use a single bors instance for all repos.

If we do not use a GitHub App, then I'd auto-derive the env like `BORS_WEBHOOK_SECRET_rust-clippy`. If we use a GitHub App, there would be a single webhook (and thus a single secret) for all the repos, with no need to deriving variable names.

At that point though we could just configure everything with environment variables, as the list of repositories bors should care about could be fetched from GitHub (list of installed apps).

Reply:
FWIW the webhook could just be a global one across the org (triagebot does that). But I think preferring the GitHub app is definitely better.